src/molecule_podman/playbooks/create.yml

```diff
@@ -8,16 +8,13 @@
   vars:
     podman_exec: "{{ lookup('env','MOLECULE_PODMAN_EXECUTABLE')|default('podman',true) }}"
   tasks:
-    - name: get podman executable path
-      ansible.builtin.command: which {{ podman_exec }}
-      register: podman_path
-      environment:
-        PATH: "{{ ansible_env.PATH }}:/sbin:/usr/sbin"
-      changed_when: false
-
-    - name: save path to executable as fact
+    - name: Register podman executable path
       ansible.builtin.set_fact:
-        podman_cmd: "{{ podman_path.stdout }}"
+        podman_cmd: "{{ lookup('ansible.builtin.first_found', params) }}"
+      vars:
+        params:
+          files: "{{ [podman_exec] }}"
+          paths: "{{ (lookup('env', 'PATH') | split(':')) + ['/sbin', '/usr/sbin'] }}"
```

Comment:
Both works.
But, I prefer installing `which` to make the logic in playbook concise.

Reply:
I didn't know about the maintenance issue. If the things are kept this way, I think that using either ``which`` or ``ansible.builtin.find`` or any other module is better. While this playbook is run on localhost, so a module or a lookup will end up doing the same, I think that lookup is to be avoided. afair lookup are run on the controller and the module is on the host so semantically the two things are very different.

(I'm also surprised it's looking at ``*/sbin`` as on my SLE and Debian systems it's in ``/bin`` but it has nothing to do with this PR)

Reply:
I feel like a single `set_fact`  is actually making the logic in the playbook more concise than having both a `command` and `set_fact` tasks, and imho molecule drivers should be highly portable and external dependencies should be avoided as much as possible.

@apatard I'm not sure that I understand how a lookup vs. a module would make any difference in this scenario, especially since lookups are being used in almost every other task of the playbook. Care to shed some light on your concerns?

BTW, the `which` task isn't even working correctly as is, `{{ ansible_env.PATH }}` isn't being populated since fact's aren't being gathered, so it's essentially just looking for the `podman_exec` under `/sbin` and `/usr/sbin`

Perhaps a better solution would be to simply stop trying to discover the path to `podman_exec` and instead require that you define the executable override with a full path.
```diff
- podman_exec: podman
+ podman_exec: /usr/bin/podman
```